app/Models/WorkshopBalance.php

```diff
@@ -58,7 +65,7 @@ public static function generateBalances($semester_id)
             $not_yet_paid = 0;
             foreach ($workshop->users as $member) {
                 if (isset($active_users[$member->id])) {
-                    if (!isset($users_has_to_pay_kktnetreg[$member->id])) {
+                    if (!is_null($member->payedKKTInSemester(Semester::find($semester_id)))) {
```

Comment:
Get the semester outside of the loop. Or even outside this function and get the semester model as param. It makes our type checking stronger.